extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/serde/tableExtractors.scala

```diff
@@ -145,14 +175,37 @@ class DataSourceV2RelationTableExtractor extends TableExtractor {
         val maybeCatalogPlugin = invokeAs[Option[AnyRef]](v2Relation, "catalog")
         val maybeCatalog = maybeCatalogPlugin.flatMap(catalogPlugin =>
           lookupExtractor[CatalogPluginCatalogExtractor].apply(catalogPlugin))
-        val maybeIdentifier = invokeAs[Option[AnyRef]](v2Relation, "identifier")
```

Comment:
The `identifier` in `DataSourceV2Relation` is not reliable enough. In fact, datasourceV2 will parse out the table based on identifier, so table is more appropriate here.
For example, in iceberg `rewrite_data_files` command, `identifier` is rewrite task id, iceberg catalog will load table by it ( can see in `SparkCatalog#load` the `isPathIdentifier` function)

Reply:
Oh, got your point. Iceberg procedure rewrites this identifier, and makes it irrelevant to the original table.

Reply:
I think we need to `TableTableExtractor` for taking `Table` input, instead of use `StringTableExtractor` directly.

Reply:
Please attach the evidence showing the impacts and differences in logical plan details of Iceberg call procedure.

Reply:
When CALL ice.system.rewrite_data_files command does the rewrite operation, two logical plans will be generated, for example in my unit test:

> 
  * This triggers two logical plans( input-files(2) >= min-input-files(2) ):
   *
   * == Physical Plan 1 ==
   * (1) Call
   *
   * == Physical Plan 2 ==
   * AppendData (3)
   * +- * ColumnarToRow (2)
   * +- BatchScan local.iceberg_ns.call_command_table (1)

 In the second logical plan, the `DataSourceV2Relation#identifier` in `AppendData` will be rewrite task id. So it is a more appropriate way to parse the table name from `DataSourceV2Relation#table`
 
<img width="1318" alt="image" src="https://github.com/apache/kyuubi/assets/31469905/efc815cb-8dfe-4a9f-8e24-d62883ef6eff">